public/app/features/panel/panel_directive.ts

```diff
@@ -6,7 +6,7 @@ import baron from 'baron';
 const module = angular.module('grafana.directives');
 
 const panelTemplate = `
-  <div class="panel-container" ng-class="{'panel-container--no-title': !ctrl.panel.title}">
+  <div class="panel-container" ng-class="{'panel-container--no-title': !ctrl.panel.title.trim()}">
```

Comment:
this messes with the watcher, never call functions in angular bindings. Don't think we need to care about white space in the title, if someone has entered that then it's not empty. Possibly add trimming on the save step / form input. But not the render path

Reply:
>this messes with the watcher, never call functions in angular bindings

Didn't know about it, thanks. Wanted to make sure that '  ' is interpreted as empty title. Can't find a good way to do that on panel options save (for angular panels)